src/order/upper_lower.lean

```diff
@@ -0,0 +1,295 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Sara Rousta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Sara Rousta
+-/
+import data.set.lattice
+import data.set_like.basic
+
+/-!
+# Up-sets and down-sets
+
+This file defines upper and lower sets in an order.
+
+## Main declarations
+
+* `is_upper_set`: Predicate for a set to be an upper set. This means every element greater than a
+  member of the set is in the set itself.
+* `is_lower_set`: Predicate for a set to be a lower set. This means every element less than a member
+  of the set is in the set itself.
+* `upper_set`: The type of upper sets.
+* `lower_set`: The type of lower sets.
```

Comment:
TODO: The bijection between upper sets and antichains, and the relationship of lattice structures on both.